src/routing/approved.rs

```diff
@@ -678,7 +784,13 @@ impl Approved {
     // If elder, always handle UserMessage, otherwise handle it only if addressed directly to us
     // as a node.
     fn should_handle_user_message(&self, dst: &DstLocation) -> bool {
-        self.is_elder() || dst.as_node().ok() == Some(&self.node.name())
+        let is_elder = self.is_elder();
+        let are_we_dst = if let DstLocation::Node(name) = dst {
```

Comment:
better having a `as_node` function as previous?

or can we do:  `DstLocation::Node(self.node.name()) == dst` ?

Reply:
> DstLocation::Node(self.node.name()) == dst

This seems better IMO.